submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/parser/TensorboardSpecParser.java

```diff
@@ -0,0 +1,154 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+public class TensorboardSpecParser {
+  public static V1Deployment parseDeployment(String name, String image, String route_path, String pvc) {
+
+    /*
+    [start] Deployment
+     */
+    V1Deployment deployment = new V1Deployment();
+
+    // [start] deployment - metadata
+    V1ObjectMeta deployment_metedata = new V1ObjectMeta();
+    deployment_metedata.setName(name);
+    deployment.setMetadata(deployment_metedata);
+    // [end] deployment - metadata
+
+    // [start] deployment - spec
+    V1DeploymentSpec deployment_spec = new V1DeploymentSpec();
+    deployment_spec.setSelector(
+        new V1LabelSelector().matchLabels(Collections.singletonMap("app", name)) // match the template
+    );
+
+    // [start] deployment - spec - template
+    V1PodTemplateSpec deployment_template_spec = new V1PodTemplateSpec();
+    deployment_template_spec.setMetadata(
+        new V1ObjectMeta().labels(Collections.singletonMap("app", name)) // bind to replicaset and service
+    );
+
+    // [start] deployment - spec - template - podspec
+    V1PodSpec deployment_template_pod_spec = new V1PodSpec();
+
+    // [start] deployment - spec - template - podspec - container
+    V1Container container = new V1Container();
+    container.setName(name);
+    container.setImage(image);
+    container.setCommand(Arrays.asList(
+        "tensorboard", "--logdir=/logs",
+        String.format("--path_prefix=%s", route_path)
+    ));
+    container.setImagePullPolicy("IfNotPresent");
+    container.addPortsItem(new V1ContainerPort().containerPort(6006));
+    container.addVolumeMountsItem(new V1VolumeMount().mountPath("/logs").name("volume"));
+    deployment_template_pod_spec.addContainersItem(container);
+    // [end] deployment - spec - template - podspec - container
+
+    // [start] deployment - spec - template - podspec - volume
+    V1Volume volume = new V1Volume().name("volume");
+    volume.setPersistentVolumeClaim(
+        new V1PersistentVolumeClaimVolumeSource().claimName(pvc)
+    );
+    deployment_template_pod_spec.addVolumesItem(volume);
+    // [end] deployment - spec - template - podspec - volume
+
+    deployment_template_spec.setSpec(deployment_template_pod_spec);
+    // [end] deployment - spec - template - podspec
+
+    deployment_spec.setTemplate(deployment_template_spec);
+    // [end] deployment - spec - template
+
+    deployment.setSpec(deployment_spec);
+    // [end] deployment - spec
+    return deployment;
+  }
+
+  public static V1Service parseService(String svc_name, String pod_name) {
+    V1Service svc = new V1Service();
+    svc.metadata(new V1ObjectMeta().name(svc_name));
+
+    V1ServiceSpec svc_spec = new V1ServiceSpec();
+    svc_spec.setSelector(Collections.singletonMap("app", pod_name)); // bind to pod
+    svc_spec.addPortsItem(new V1ServicePort().protocol("TCP").targetPort(new IntOrString(6006)).port(8080));
```

Comment:
You could create constant variables in `TensorboardUtils`  
`final String DEFAULT_TENSORBOARD_PORT = "6006";`
`final String EXPOSE_TENSORBOARD_PORT= "8080";`